In [1]:
import pandas as pd
data_frame_of_purchases = pd.read_csv('ArtHistBooks.csv')
data_frame_of_purchases
number_of_purchases = data_frame_of_purchases.shape[0]
print(f'{number_of_purchases} involved an art book, a history book, a coffee-table book, and/or our book of photos from 20th century.')

1000 involved an art book, a history book, a coffee-table book, and/or our book of photos from 20th century.


In [2]:
data_frame_of_purchases_of_art_book = data_frame_of_purchases[data_frame_of_purchases['ArtBooks'] > 0]
number_of_purchases_of_art_book = data_frame_of_purchases_of_art_book.shape[0]
print(f'{number_of_purchases_of_art_book} purchases involved an art book')

301 purchases involved an art book


In [3]:
data_frame_of_purchases_of_book_of_photos_from_20th_century = data_frame_of_purchases[data_frame_of_purchases['Purchase'] > 0]
number_of_purchases_of_book_of_photos_from_20th_century = data_frame_of_purchases_of_book_of_photos_from_20th_century.shape[0]
print(f'{number_of_purchases_of_book_of_photos_from_20th_century} purchases involved our book of photos from 20th century.')

89 purchases involved our book of photos from 20th century.


In [4]:
data_frame_of_purchases_of_art_book_and_book_of_photos_from_20th_century = data_frame_of_purchases_of_art_book[data_frame_of_purchases_of_art_book['Purchase'] > 0]
number_of_purchases_of_art_book_and_book_of_photos_from_20th_century = data_frame_of_purchases_of_art_book_and_book_of_photos_from_20th_century.shape[0]
number_of_purchases_of_art_book_and_book_of_photos_from_20th_century
print(f'{number_of_purchases_of_art_book_and_book_of_photos_from_20th_century} purchases involved an art book and our book of photos from 20th century.')

58 purchases involved an art book and our book of photos from 20th century.


In [5]:
probability_that_purchase_will_involve_book_of_photos_from_20th_century = number_of_purchases_of_book_of_photos_from_20th_century / number_of_purchases
probability_that_purchase_will_involve_book_of_photos_from_20th_century_given_purchase_involves_art_book = number_of_purchases_of_art_book_and_book_of_photos_from_20th_century / number_of_purchases_of_art_book
print(f'The probability that a purchase will involve our book of photos from 20th century P(book of photos from 20th century) = {probability_that_purchase_will_involve_book_of_photos_from_20th_century}')
print(f'The probability that a purchase will involve our book of photos from 20th century given that the purchase involves an art book P(book of photos from 20th century | art book) = {probability_that_purchase_will_involve_book_of_photos_from_20th_century_given_purchase_involves_art_book}')

The probability that a purchase will involve our book of photos from 20th century P(book of photos from 20th century) = 0.089
The probability that a purchase will involve our book of photos from 20th century given that the purchase involves an art book P(book of photos from 20th century | art book) = 0.19269102990033224


Is a customer more likely to buy our book of photos from 20th century given that we know that the customer bought an art book? Yes. The probability that a customer will purchase our book of photos from the 20th century given that the customer purchased an art book, $0.193$, is about double the probability that a customer will purchase our book of photos from 20th century, $0.089$.